src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -88,103 +128,1169 @@ static DamageTable D_80A31B1C = {
+            EnSlime_ReactToDamage(this, play, false);
+        } else {
+            this->actor.world.rot.y = this->actor.shape.rot.y;
+            EnSlime_ChangeDirectionIdle(this);
+        }
+    }
+}
 
-extern UNK_TYPE D_060004C0;
-extern UNK_TYPE D_06000828;
+/**
+ * Setup ice block unfreeze
+ * Set action function to UpdateIceBlockUnfreeze
+ */
+void EnSlime_InitIceBlockUnfreeze(EnSlime* this) {
+    this->actor.colorFilterTimer = 0;
+    this->actor.gravity = -2.0f;
+    Actor_SetColorFilter(&this->actor, 0x8000, 0x80FF, 0x2000, 10);
+    this->actionFunc = EnSlime_UpdateIceBlockUnfreeze;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/EnSlime_Init.s")
+/**
+ * Update the actor's ice block timer to match state of ice block.
+ * When it reaches the "unset" value (255), return actor to idle.
+ */
+void EnSlime_UpdateIceBlockUnfreeze(EnSlime* this, PlayState* play) {
+    s32 iceBlockTimerAdj;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/EnSlime_Destroy.s")
+    this->actor.colorFilterTimer = 10;
+    if ((this->actor.child != NULL) && (this->actor.child->update != NULL)) {
+        iceBlockTimerAdj = (0.1f - this->actor.child->scale.y) * 10.0f * 255.0f;
+        this->iceBlockTimer = CLAMP(iceBlockTimerAdj, 0, 255);
+    } else {
+        this->actor.child = NULL;
+        iceBlockTimerAdj = this->iceBlockTimer + 10;
+        this->iceBlockTimer = CLAMP_MAX(iceBlockTimerAdj, 255);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2EFAC.s")
+    if (this->iceBlockTimer == 255) {
+        this->collider.base.acFlags |= AC_ON;
+        this->actor.flags |= ACTOR_FLAG_1;
+        this->actor.flags &= ~ACTOR_FLAG_10;
+        EnSlime_ReturnToIdle(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F028.s")
+/**
+ * Set speed to zero, and make undrawable.
+ * Initialize respawn timer.
+ * Set action function to CountdownRespawn
+ */
+void EnSlime_HideDead(EnSlime* this) {
+    this->actor.draw = NULL;
+    this->actor.flags |= ACTOR_FLAG_10;
+    this->effectAlpha = 0.0f;
+    this->actor.gravity = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->actor.speed = 0.0f;
+    this->collider.base.ocFlags1 &= ~OC1_ON;
+    this->timer = this->respawnTime;
+    this->actionFunc = EnSlime_CountdownRespawn;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F0A8.s")
+/**
+ * Decrement the respawn timer, or if it is zero,
+ * initialize respawn action cycle.
+ */
+void EnSlime_CountdownRespawn(EnSlime* this, PlayState* play) {
+    if (this->timer == 0) {
+        EnSlime_BeginRespawn(this);
+    } else {
+        this->timer--;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F110.s")
+/**
+ * Re-enable drawing, reset the actor's health and home,
+ * and setup and initialize respawn action.
+ * Set action function to UpdateRespawn
+ */
+void EnSlime_BeginRespawn(EnSlime* this) {
```

Comment:
Similarly to above, EnBb calls the equivalent to these functions `EnBb_SetupRevive` and `EnBb_Revive`. What do you think? I think "Respawn" works too, just make sure that all four functions are consistent with whichever name you choose.